user-interface/src/data-verification/ConsolidationOrderModal.tsx

```diff
@@ -33,22 +33,46 @@ export interface ConsolidationOrderModalProps {
   onConfirm: (results: ConfirmActionResults) => void;
 }
 
+type ConfirmationSteps = 'pick-lead-case' | 'confirm';
+
 type ShowOptionParams = {
   status: OrderStatus;
-  caseIds: string[];
-  attorneys: AttorneyInfo[];
+  cases: ConsolidationOrderCase[];
 };
 
 type ShowOptions = {
   status: OrderStatus;
   heading: string;
-  attorneys: AttorneyInfo[];
 };
 
 export type ConfirmationModalImperative = ModalRefType & {
   show: (options: ShowOptionParams) => void;
 };
 
+export async function getCaseAssignments(caseId: string): Promise<Array<CaseAssignment>> {
+  const api = useApi();
+  const response = await api.get(`/case-assignments/${caseId}`);
+  return response.body as CaseAssignment[];
+}
+
+export async function fetchLeadCaseAttorneys(leadCaseId: string): Promise<Array<string>> {
+  const assignments = await getCaseAssignments(leadCaseId);
+  const attorneys = assignments.map((assignment) => assignment.name);
+  return attorneys;
+}
+
+export function addOxfordCommas(attorneys: string[]) {
+  const newAttorneyList = [...attorneys];
```

Comment:
We can defer the array copy until we get past the first two length === 0 and length < 3 conditions.